Source/Core/CoplanarPolygonGeometryLibrary.js

```diff
@@ -0,0 +1,131 @@
+define([
+        './defined',
+        './Cartesian2',
+        './Cartesian3',
+        './Check',
+        './IntersectionTests',
+        './Math',
+        './Matrix3',
+        './OrientedBoundingBox',
+        './Plane',
+        './Ray'
+    ], function(
+        defined,
+        Cartesian2,
+        Cartesian3,
+        Check,
+        IntersectionTests,
+        CesiumMath,
+        Matrix3,
+        OrientedBoundingBox,
+        Plane,
+        Ray
+    ) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    var PolygonGeometryLibrary = {};
+
+    var scratchIntersectionPoint = new Cartesian3();
+    var scratchXAxis = new Cartesian3();
+    var scratchYAxis = new Cartesian3();
+    var scratchZAxis = new Cartesian3();
+    var scratchOrigin = new Cartesian3();
+    var scratchNormal = new Cartesian3();
+    var scratchRay = new Ray();
+    var scratchPlane = new Plane(Cartesian3.UNIT_X, 0);
+    var obbScratch = new OrientedBoundingBox();
+
+    // call after removeDuplicates
+    PolygonGeometryLibrary.projectTo2D = function(positions, positionsResult, normalResult, tangentResult, bitangentResult) {
```

Comment:
I think this is cleaner than returning an object, especially because the normal/tangent/bitangent are only used for the filled polygon, not the outline.

Reply:
This is a pretty common pattern in c++ programming where you can also pass by reference